javascript/ql/src/LanguageFeatures/InconsistentNew.ql

```diff
@@ -95,8 +95,20 @@ predicate whitelistedCall(DataFlow::CallNode call) {
   exists(call.getReceiver())
 }
 
+/**
+ * Get the `new` or call (depending on whether `isNew` is true or false) of `f`
+ * that comes first under a lexicographical ordering by file path, start line
+ * and start column.
+ */
+DataFlow::InvokeNode getFirstInvocation(Function f, boolean isNew) {
+  result = min(DataFlow::InvokeNode invk, string path, int line, int col |
+    f = getALikelyCallee(invk, isNew) and invk.hasLocationInfo(path, line, col, _, _) |
+    invk order by path, line, col
+  )
+}
+
 from Function f, DataFlow::NewNode new, DataFlow::CallNode call
-where f = getALikelyCallee(new, true) and
-      f = getALikelyCallee(call, false)
+where new = getFirstInvocation(f, true) and
```

Comment:
Could we show the number of unlisted calls like: 

```
F is invoked as a constructor _here_ (and 17 other locations), and as a normal function _here_ (and 2 other locations) 
```
or
```
F is invoked as a constructor at 17 locations (such as _here_), and as a normal function at 2 locations (such as _here_)
```

I think being explicit in the query message about not showing all results will make the programmer happier than only alerting him once the query is run again.

Reply:
I agree about making the query message more explicit, but adding the count is a non-trivial change. How about "F is sometimes invoked as a constructor (for example _here_) and sometimes as a normal function (for example _here_)"?

Reply:
Agreed.